aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -130,7 +130,7 @@ def _item_to_data_record(self, item: Dict[str, Any]) -> DataRecord:
             idempotency_key=item[self.key_attr],
             status=item[self.status_attr],
             expiry_timestamp=item[self.expiry_attr],
-            response_data=item.get(self.data_attr),
+            response_data=item.get(self.data_attr, ""),
```

Comment:
Could you document what this addresses? Concern: besides type annotations, we're changing code that might have been a subtle bug OR introducing a bug we haven't covered in tests either -- it's hard to know and review.

Reply:
So the reasoning is as following:
Previously we allowed `response_data` to be None. In class itself we define response data as
`response_data: Optional[str] = ""`. Which means that if we don't provide this value we get empty string, but if we provide a None( as with line 133) it will be accepted. This will break if user use `response_json_as_dict` method https://github.com/awslabs/aws-lambda-powertools-python/blob/0196883410bf4fadeb1ca27f3f331ec6fbaffbe4/aws_lambda_powertools/utilities/idempotency/persistence/base.py#L95 with `TypeError: the JSON object must be str, bytes or bytearray, not NoneType` . Because of that I decided to set response_data to empty string instead of None here.

Different approach I can take is to change behaviour of `response_json_as_dict` and validate if input is not None and only then try to create json from it. What do you think @heitorlessa ?

Reply:
Gotcha!! It'd be best to handle that within `response_json_as_dict` because if we add another persistence layer like Redis then we're solved ;)

Reply:
Yep, make sense to solve it once and for all :-)

```
    def response_json_as_dict(self) -> Optional[dict]:
        """
        Get response data deserialized to python dict

        Returns
        -------
        Optional[dict]
            previous response data deserialized
        """
        return json.loads(self.response_data) if self.response_data else None
```